File: backend/passwordRecovery.go
package main
//necessary libraries
import(
	"fmt"
	"math/rand"
	"net/smtp"
)
//function below is for making a new random password with length of 10
func passwordGenerator() string{
	randomPassword := ""
	characters := "abcdefghijklmnopqrstuvwxyz1234567890!@#$%^&*()"
	charactersLength := len(characters)
	for i := 0;i < 10;i++ {
		randomIndex := rand.Intn(charactersLength)
		randomCharacter := characters[randomIndex:randomIndex+1]
		randomPassword += randomCharacter
	}
	return randomPassword
}
//function below sends recovery email to target main Address
func sendEmail(targetEmail string){
	//sender email properties
	from := "mmdspam79@gmail.com"
	emailPassword := "fshvlvsknwthnxrf"
	randomPassword := passwordGenerator()
	randomPasswordBytes := []byte(randomPassword)
	//host and port
	host := "smtp.gmail.com"
	port := "587"
	//sending email to targets
	toSendList := []string{targetEmail}
	authentication := smtp.PlainAuth("",from,emailPassword,host)
	error := smtp.SendMail(host+":"+port,authentication,from,toSendList,randomPasswordBytes)
	fmt.Println(error)
}
//main function
func main(){
	//TODO
}

File: backend/passwordRecovery.py
#importing necessary libraries
import smtplib
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText
from search import *
#target mail is changed based on conditions
targetMail = 'allo@gmail.com'
mailContent = searchPassword(targetMail)#content is password of the target
#test
print(mailContent)
#reciever and target mails
senderMail = 'mmdspam79@gmail.com'
senderPassword = 'fshvlvsknwthnxrf'
receiverMail = 'meymani79@gmail.com'#it should be target mail here we just want to test
#Setup the MIME protocol
message = MIMEMultipart()
message['From'] = senderMail
message['To'] = receiverMail
message['Subject'] = 'YOUR PASSWORD'
#The body and the attachments for the mail
message.attach(MIMEText(mailContent, 'plain'))
#Create SMTP session for sending the mail
session = smtplib.SMTP('smtp.gmail.com', 587) #host and port
session.starttls() #enable security
session.login(senderMail, senderPassword)
text = message.as_string()
session.sendmail(senderMail, receiverMail, text)
session.quit()
print('Mail Sent')